Add mentor/mentee availability option on Sign Up

haroldadmin opened the pull request:
### Description
Modified the activity layout to include two additional checkboxes for Mentor/Mentee availability, and also added these two state variables to the RegisterRequest data class. The SignUpActivity now passes these two parameters to the register request, based on the state of the check boxes.

Fixes #64 

### Type of Change:

- Code
- User Interface

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

## Screenshot of changes:
![mentor_mentee_feature](https://user-images.githubusercontent.com/24315306/47496828-acff8400-d875-11e8-94b9-e32a4633dbc5.jpg)

### How Has This Been Tested?
Tested sign up flow on personal device

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or material

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

Rimjhim28 reviewed (commented):
Hey @haroldadmin, you should actually squash the 2 commits so that we can have a clean commit history when your PR is merged. :D

isabelcosta commented:
@haroldadmin Can you please do the following:
- [x] squash all commits just like @Rimjhim28 said
- [x] follow the [Commit Message Style Guide](https://github.com/systers/mentorship-android/wiki/Commit-Message-Style-Guide). E.g.: name commit `feat: add mentor and mentee availability option to sign up`
- [x] add a visual component to the PR description, a GIF or image works (the ones you presented on slack work) In this way anyone can see what your PR adds to the Registration UI

isabelcosta commented:
@haroldadmin a good practise is to create a branch per each feature/bug fix... issue you work on. I see you used develop branch to work on this issue. If you start work on another issue, you won't want to have this PR's changes (if not merged yet).

haroldadmin commented:
@isabelcosta I'm sorry for this messy PR, I'm very new to FOSS contribution 😅. 
I'll complete the requested changes and update the PR soon!

haroldadmin commented:
@isabelcosta @Rimjhim28 I have made the requested changes, please review and provide feedback.

Also, @isabelcosta I will make sure that I do my work in separate branches from now on, thanks for the idea 😊

isabelcosta commented:
@haroldadmin I was new to FOSS  as well a few months ago, what matters is that you learn from this experience. No problem if you don't do it completely according to our guidelines at first, now that I asked you some changes to the PR and you did it, you learned something for sure!

Thank you for following our guidelines, when I get some time I'll do a proper review of your work!

haroldadmin commented:
I identified an issue with my last commit, where I was setting the values of `needsMentoring` and `isAvailableToMentor` equal to `isActivated` state of checkboxes, instead of `isChecked`. I have created a new commit that addresses this issue. I have verified my changes by signing up with a new account with checkboxes checked. After logging in with the new account, the ProfileFragment screen shows the switches as activated, which verifies that the system is indeed working properly.

isabelcosta commented:
@haroldadmin So glad you noticed that bug yourself. Real nice catch 🎉

isabelcosta commented on app/src/main/java/org/systers/mentorship/remote/requests/RegisterRequest.kt:
```diff
@@ -16,5 +16,7 @@ data class RegisterRequest (
         val password: String,
-        @SerializedName("terms_and_conditions_checked") val hasAcceptedTermsAndConditions: Boolean
+        @SerializedName("terms_and_conditions_checked") val hasAcceptedTermsAndConditions: Boolean,
+		@SerializedName("need_mentoring") val needsMentoring: Boolean,
```
Fix indentation please

haroldadmin replied:
Fixed in the latest commit. Nice catch! It showed up as a well formatted string in my IDE, so it must have slipped past me. I have formatted it properly now.

isabelcosta commented on app/src/main/res/layout/activity_sign_up.xml:
```diff
@@ -3,41 +3,38 @@
 
         <android.support.v7.widget.AppCompatTextView
-            android:id="@+id/textView"
+            android:id="@+id/signUpTextview"
```
rename this ID to `tvSignUp` just to follow the rest of the xml elements ids current convention

haroldadmin replied:
Fixed in the latest commit.

isabelcosta reviewed (changes requested):
Great job 👍

haroldadmin committed:
feat: add mentor and mentee availability option to sign up

Fixes issue #64 by adding two checkboxes to indicate availability as
Mentor/Mentee while signing up. Modifes the activity layout, as well
RegisterRequest data class, SignUpActivity to include these two state
variables in the register-request to the server.

isabelcosta reviewed (approved):
Thank you for doing the requested changes :) Great work @haroldadmin
